### PR TITLE
test(frontend): add unit tests for DaemonSet and StatefulSet model classes

### PR DESCRIPTION
**Pull request by srajang1805**
## Summary

Adds comprehensive unit tests for the DaemonSet and StatefulSet model classes following the existing `pod.test.ts` testing patterns.

### Changes

* Added `frontend/src/lib/k8s/daemonSet.test.ts`

  * Tests `getBaseObject()`
  * Tests `getContainers()`
  * Tests `getNodeSelectors()`
  * Tests `getCurrentRevision()`
  * Covers missing spec/template/nodeSelector/generation edge cases

* Added `frontend/src/lib/k8s/statefulSet.test.ts`

  * Tests `getBaseObject()`
  * Tests `getContainers()`
  * Tests `getCurrentRevision()`
  * Tests `StatefulSet.isScalable`
  * Covers missing spec/template/generation edge cases

### Test Coverage

* 11 DaemonSet model tests
* 9 StatefulSet model tests
* 41/41 model tests passing across:

  * Pod
  * Service
  * DaemonSet
  * StatefulSet

### Notes for Reviewers

* Follows the existing `pod.test.ts` structure and conventions.
* Pre-commit hooks, linting, and test suite pass successfully.

### CODE REVIEW

**Comment by srajang1805** [comment]
@illume PR ready for re-review.

**Review by illume — CHANGES_REQUESTED** [review]
Thanks for these changes.


Can you please address the open review comments? Once you've resolved each one, please mark it as resolved.

File: frontend/src/lib/k8s/statefulSet.test.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      const base = StatefulSet.getBaseObject();
+      expect(base.spec.template.spec.containers).toHaveLength(1);
+      expect(base.spec.template.spec.containers![0].name).toBe('');
+      expect(base.spec.template.spec.containers![0].image).toBe('');
+      expect(base.spec.template.spec.containers![0].imagePullPolicy).toBe('Always');
+    });
+
+    it('has isScalable static property set to true', () => {
+      expect(StatefulSet.isScalable).toBe(true);
+    });
```
[the remark this answers is not in the document]
Comment (srajang1805) [reply frontend/src/lib/k8s/statefulSet.test.ts:82]: PR desc changed as requested

**Comment by srajang1805** [comment]
@illume Ready for review

**Review by illume — APPROVED** [review]
🎉 thanks!

**Comment by srajang1805** [comment]
@illume  Thanks!
